lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -76,6 +76,8 @@ def __init__(self, b_working_directory, validate_certs=True):
         self._galaxy_collection_cache = {}  # type: Dict[Union[Candidate, Requirement], Tuple[str, str, GalaxyToken]]
         self._b_working_directory = b_working_directory  # type: bytes
 
+        self.keyring = keyring
```

Comment:
Maybe make this `_keyring` and expose `keyring` as a read-only property?

Reply:
Done